src/sentry/static/sentry/app/utils/consolidatedScopes.jsx

```diff
@@ -15,123 +15,105 @@ const HUMAN_RESOURCE_NAMES = {
-   *
-   * Example:
-   *    Given the full list of scopes:
-   *      ['project:read', 'project:write', 'team:read', 'team:write', 'team:admin']
-   *
-   *    this would return:
-   *      ['project:write', 'team:admin']
-   */
-  topScopes(scopeList) {
-    return Object.values(groupBy(scopeList, scope => scope.split(':')[0]))
-      .map(scopes => scopes.sort(this.compareScopes))
-      .map(scopes => scopes.pop());
+/**
+ * Convert into a list of Permissions, grouped by resource.
+ *
+ * This is used in the new/edit Sentry App form. That page displays permissions
+ * in a per-Resource manner, meaning one row for Project, one for Organization, etc.
+ *
+ * This exposes scopes in a way that works for that UI.
+ *
+ * Example:
+ *    {
+ *      'Project': 'read',
+ *      'Organization': 'write',
+ *      'Team': 'no-access',
+ *      ...
+ *    }
+ */
+function toResourcePermissions(scopes) {
+  const permissions = {...DEFAULT_RESOURCE_PERMISSIONS};
+  let filteredScopes = [...scopes];
+  // The scope for releases is `project:releases`, but instead of displaying
+  // it as a permission of Project, we want to separate it out into its own
+  // row for Releases.
+  if (scopes.includes(PROJECT_RELEASES)) {
+    permissions.Release = 'admin';
+    filteredScopes = scopes.filter(scope => scope !== PROJECT_RELEASES); // remove project:releases
   }
 
-  compareScopes = (a, b) => {
-    return this.permissionLevel(a) - this.permissionLevel(b);
-  };
+  topScopes(filteredScopes).forEach(scope => {
+    const [resource, permission] = scope.split(':');
+    permissions[HUMAN_RESOURCE_NAMES[resource]] = permission;
+  });
 
-  /**
-   * Numerical value of the scope where Admin is higher than Write,
-   * which is higher than Read. Used to sort scopes by access.
-   */
-  permissionLevel = scope => {
-    const permission = scope.split(':')[1];
-    return PERMISSION_LEVELS[permission];
-  };
+  return permissions;
+}
 
-  get defaultResourcePermissions() {
-    return {
-      Project: 'no-access',
-      Team: 'no-access',
-      Release: 'no-access',
-      Event: 'no-access',
-      Organization: 'no-access',
-      Member: 'no-access',
-    };
-  }
+/**
+ * Convert into a list of Permissions, grouped by access and including a
+ * list of resources per access level.
+ *
+ * This is used in the Permissions Modal when installing an App. It displays
+ * scopes in a per-Permission way, meaning one row for Read, one for Write,
+ * and one for Admin.
+ *
+ * This exposes scopes in a way that works for that UI.
+ *
+ * Example:
+ *    {
+ *      read:  ['Project', 'Organization'],
+ *      write: ['Member'],
+ *      admin: ['Release']
+ *    }
+ */
+function toPermissions(scopes) {
+  const defaultPermissions = {read: [], write: [], admin: []};
+  const resourcePermissions = toResourcePermissions(scopes);
+
+  // Filter out the 'no-access' permissions
+  const permissions = pick(invertBy(resourcePermissions), ['read', 'write', 'admin']);
```

Comment:
@nolachen Nice job making this functional. While `pick` certainly works in this case, if all you want to do is filter out a property then `omit` is probably the better function to use.

Reply:
`omit` is apparently considerably slower : https://lodash.com/docs/4.17.15#omit

Reply:
@nolachen Interesting about performance, that's good to know. As you said, doesn't matter here. Treat it as an optional suggestion.